src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -1,34 +1,45 @@
 package com.kodilla.ecommercee.controller;
+
+import com.kodilla.ecommercee.dto.CartDto;
 import com.kodilla.ecommercee.entity.Cart;
 import com.kodilla.ecommercee.GenericEntity;
 import com.kodilla.ecommercee.entity.UserOrder;
-
-
-
+import com.kodilla.ecommercee.exception.CartNotFoundException;
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.service.CartService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/v1/cart")
 public class CartController {
 
+    @Autowired
+    private CartService cartService;
 
+    @Autowired
+    private CartMapper cartMapper;
 
-
-    public Cart createCart(Cart cart){
-        return new Cart();
+    @PostMapping
+    public Cart createCart(CartDto cartDto){
+        return cartService.createCart(cartMapper.translateToCart(cartDto));
     }
-
-    public GenericEntity getProductFromCart(Cart cart){
-        return new GenericEntity();
+    @GetMapping(value = "/{cartId}")
+    public Optional<GenericEntity> getProductFromCart(@PathVariable Long cartId, int productId) throws CartNotFoundException {
+        return Optional.ofNullable(cartService.getProductFromCart(cartId, productId).orElseThrow(CartNotFoundException::new));
     }
-
-
-    public Cart addProductToCart(GenericEntity product){
-        return new Cart();
+    @PutMapping(value = "/{cartId}")
+    public boolean addProductToCart(@PathVariable Long cartId,GenericEntity product){
```

Comment:
Nie używajmy w ogóle tego GenericEntity. Tutaj powinno być Jako drugi parametr ProductDto.
Powinna tez być adnotacja która mówi jak to jest przekazywane. Np.
```
@RequestBody ProductDto productdto
```